rock_paper_scissors/rps.py

```diff
@@ -2,13 +2,25 @@
 
 import sys
 
+
 def rock_paper_scissors(n):
-  pass 
+    signs, results = ['rock', 'paper', 'scissors'], []
+
+    def build_permutation(n_list=[]):
+        if len(n_list) < n:
+            for sign in signs:
+                p_list = n_list[:]
+                p_list.append(sign)
+                build_permutation(p_list)
+        else:
+            results.append(n_list)
+    build_permutation()
+    return results
```

Comment:
This is a fantastic implementation Joseph ~ Good choice (for many reasons) avoiding recursion, and it is great that you solved it with relatively low performance cost. Can you think of a way to optimize it a little more?